### DISCUSSION TITLE
Photos uploaded don't show in timeline

### DISCUSSION THREAD

**Discussion by AngelPone** [immich-app/immich#6349] [Q&A] [answered]
I wrote a Python scripts to upload photos. However, some photos don't show in timeline and albums. Some photos correctly show.
I checked the assets table in database. The assets uploaded are in there.
The assets count of the album is correct and I can see the album thumbnail.
However, some assets didn't shown either in the main timeline or in the timeline of the album.
Uploading through immich cli or web are all normal.

The code related to uploading is attached:

```Python
  def upload_file(self, file):
      """Upload a single file to Immich"""
      with open(file, "rb") as f:
          form_data = get_form_data(file, f)
          resp = requests.post(self.url + f"/api/asset/upload", data = form_data, headers = self.authenticate(),
                               files = {"assetData": f})

def get_form_data(file_path):
    stats = os.stat(file_path)
    return {"deviceId": "CLI", "fileCreatedAt": datetime.fromtimestamp(stats.st_mtime).isoformat(),
            "fileModifiedAt": datetime.fromtimestamp(stats.st_mtime).isoformat(),
            "fileSize": stats.st_size, "isFavorite": "false", "deviceAssetId": f"{file_path.split('/')[-1]}-{stats.st_size}"}

          return resp.json()
```

Any idea why this happens?

**Comment by JW-CH** [comment]
Not exactly a response to your problem/question, but why do you use a python script and not the provided CLI-Tool by immich?

**Reply by AngelPone** [reply]
Currently immich CLI can not upload directly to shared album. I want to write a temporary solution.

**Reply by JW-CH** [reply, marked as answer]
> Currently immich CLI can not upload directly to shared album. I want to write a temporary solution.

Immich-go can upload into an existing album (can be shared). Maybe this would be worth to take a look at.

https://github.com/simulot/immich-go
